src/btree/bt_split.c

```diff
@@ -743,19 +743,21 @@ __split_parent(WT_SESSION_IMPL *session, WT_REF *ref, WT_REF **ref_new,
 	 * the WT_REF being replaced might have been in a WT_REF_DELETED state,
 	 * in which case it was switched to a WT_REF_SPLIT state and counted as
 	 * a deleted entry, don't double count it.
+	 *
+	 * We can't leave an empty internal page: if there are no remaining
+	 * entries on the parent, leave the first element in place and mark
+	 * the page to be evicted soon.
 	 */
 	result_entries = (parent_entries + new_entries) - deleted_entries;
 	if (ref->state != WT_REF_SPLIT)
 		--result_entries;
-
-	/*
-	 * If there are no remaining entries on the parent, give up, we can't
-	 * leave an empty internal page. Mark it to be evicted soon and clean
-	 * up the references that have changed state.
-	 */
 	if (result_entries == 0) {
+		pindex->index[0]->state = WT_REF_DELETED;
```

Comment:
Right; and if there's only a single entry, it's even trickier. I think we could make it work, but I don't see it's a big deal (mostly because we'll only fail for this reason once, it won't happen repeatedly), so I'll close this branch as well.